assets/js/workflow-reducer.js

```diff
@@ -122,6 +123,26 @@ registerReducerFunc(RELOAD_WORKFLOW + '_FULFILLED', (state, action) => {
   });
 });
 
+// SET_WORKFLOW_PUBLIC
+// Set the workflow to public or private
+export function setWorkflowPublicAction(workflowId, isPublic) {
+  return function (dispatch) {
+    return (
+
+      dispatch({
+        type: SET_WORKFLOW_PUBLIC,
+        payload: api.setWorkflowPublic(workflowId, isPublic)
+      }).then( () => {
+
+      dispatch(
+        reloadWorkflowAction()
```

Comment:
Why is the reload necessary? Isn't the public flag in the redux store, and therefore ultimately passed as a prop to the navbar where it's displayed?

Reply:
It may not be necessary, but we have to deal with both 'public' and 'isReadOnly', and the way that both 'public' and 'readOnly' properties are handled needs a full re-write that I wasn't ready to deal with here.